ginga/misc/plugins/Cuts.py

```diff
@@ -319,23 +330,54 @@ def stop(self):
 
     def _plotpoints(self, obj, color):
         image = self.fitsimage.get_image()
-        # Get points on the line
-        if obj.kind == 'line':
-            points = image.get_pixels_on_line(int(obj.x1), int(obj.y1),
-                                              int(obj.x2), int(obj.y2))
-        elif obj.kind in ('path', 'freepath'):
-            points = []
-            x1, y1 = obj.points[0]
-            for x2, y2 in obj.points[1:]:
-                pts = image.get_pixels_on_line(int(x1), int(y1),
-                                               int(x2), int(y2))
-                # don't repeat last point when adding next segment
-                points.extend(pts[:-1])
-                x1, y1 = x2, y2
-
-        points = numpy.array(points)
-        self.plot.cuts(points, xtitle="Line Index", ytitle="Pixel Value",
-                       color=color)
+
+        if self.cut_axis_enabled:
+            path = image.get('path', None)
+            hdus = pyfits.open(path, 'readonly')
+            phd = hdus[0].data
```

Comment:
This is a bit hacky, but I use it to open a given HDU automatically from within a plugin. This is a code snippet from a method that belongs to the plugin:
```python
class MyLocalPlugin(...):
    def redo(self):
        # stuff

        #----- Extract ERR extension -----
        # errsrc is ERR version of image
        imfile = image.metadata['path']
        imname = image.metadata['name'].split('[')[0]
        err_extnum = 2  # I have my own function to find it, but let's say it is always 2 here for simplicity
        if err_extnum < 0:
            errsrc = None
        else:
            chname = self.fv.get_channelName(self.fitsimage)
            chinfo = self.fv.get_channelInfo(chname)
            errname = '{0}[{1}]'.format(imname, err_extnum)
            if errname in chinfo.datasrc:  # ERR already loaded
                errsrc = chinfo.datasrc[errname]
            else:  # Force load ERR data
                errsrc = self.fv.load_image(imfile, idx=err_extnum)
                chinfo.datasrc[errname] = errsrc
                self.fv.make_callback('add-image', chinfo.name, errsrc)

        # more stuff
```